crates/trie/trie/src/hashed_cursor/post_state.rs

```diff
@@ -22,7 +22,7 @@ impl<'a, CF: HashedCursorFactory> HashedCursorFactory for HashedPostStateCursorF
 
     fn hashed_account_cursor(&self) -> Result<Self::AccountCursor, reth_db::DatabaseError> {
         let cursor = self.cursor_factory.hashed_account_cursor()?;
-        Ok(HashedPostStateAccountCursor::new(cursor, self.post_state))
+        Ok(HashedPostStateAccountCursor::new(cursor, &self.post_state.accounts))
```

Comment:
this combined with https://github.com/paradigmxyz/reth/pull/9077 makes it possible to walk over accounts and storage tables in parallel, right?

Reply:
we already do that by holding an `Arc` over `HashedPostState`. this just allows some code clean up